mistralrs-core/src/pipeline/mod.rs

```diff
@@ -467,9 +511,19 @@ pub trait NormalModel {
             .progress_with(bar)
             .for_each(|((tensor, _), device)| {
                 if let QMatMul::Tensor(t) = tensor {
-                    n_quantized.fetch_add(1, std::sync::atomic::Ordering::Relaxed);
                     let t = t.to_device(&device).unwrap();
-                    *tensor = QMatMul::QTensor(Arc::new(QTensor::quantize(&t, dtype).unwrap()));
+                    let quantization_behaviour = get_quantization_behaviour(&t, dtype);
+                    *tensor =  match quantization_behaviour{
+                        QuantizationBehaviour::Skip => {
+                            let shape = t.shape();
+                            warn!("Skipping quantization of tensor with shape {shape:?} as it is not quantizable.");
+                            QMatMul::QTensor(Arc::new(QTensor::quantize(&t, GgmlDType::F32).unwrap()))
```

Comment:
In the case of skipping the tensor, can we avoid converting it to F32 by just returning `tensor`. For example, if you are loading on CUDA, the tensors are loaded into (cpu) memory in BF16, so converting to F32 may incur a performance and memory cost.

Reply:
Yeah if we can keep the tensors in their original `dtype` that would be great. Can we just return a normal `Tensor` here or does it need to be a `QMatMul::QTensor`? 

And is it save to keep the original `dtype`, or do we have a risk for miss-matched `dtypes` later on? (I thought the ggml matmul always returns a f32/f16 now) 🤔

Reply:
Ah looks like, the `QMatMul` also simply accepts a `Tensor` or `TensorF16` variant. Should we return the `TensorF16` if we run on a gpu or should that be handled elsewhere?

Reply:
Oh, you're right, I overlooked that. The ggml matmul always returns f32/f16 now so we'd probably get a dtype error if some other tensors had ISQ applied. I think in that case it's better to pay the cost upfront in total memory usage rather than cast at runtime, what do you think?